### ISSUE TITLE
fullWidth prop should be selected by default

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#2567]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

For all components that have a fullWidth prop, it would be better to keep it checked by default.

<img width="252" alt="Screenshot 2023-08-18 at 13 10 17" src="https://github.com/mui/mui-toolpad/assets/92228082/7b72e38c-c83a-4f9b-96d7-40b70a7d0af9">

**Comment by Janpot** [comment]
For reference, when I drag some form components on the page, this is what it would look like by default:

<img width="1437" alt="Screenshot 2023-08-29 at 10 10 02" src="https://github.com/mui/mui-toolpad/assets/2109932/ebc03036-284c-464f-849a-ef685754bd4c">

**Comment by prakhargupta1** [comment]
> this is what it would look like by default:

Yes, this doesn't look good. Clearly, some of the components by default shouldn't be full width. @oliviertassinari What do you think?

**Comment by oliviertassinari** [comment]
The default value could depend on the container. For the top of the page, it's weird, but inside a page, it makes sense to me.

Two extra thought:

1. I would want to be able to resize components at the top-level, rather than their layout to take the full width, this could be another solution.
2. When dropping component in a nester container, it doesn't have to take all the space available.

**Comment by apedroferreira** [comment]
If this is because resizing components in their non-full-width state might be unintuitive, we can try to rethink things so that that the containers don't have to take full-width, or so that the components can be resized inside a container without resizing the whole container.
